spring-boot/src/main/java/org/springframework/boot/web/embedded/tomcat/SslStoreProviderUrlStreamHandlerFactory.java

```diff
@@ -66,9 +66,12 @@ protected URLConnection openConnection(URL url) throws IOException {
 											.getKeyStore());
 						}
 						if (TRUST_STORE_PATH.equals(url.getPath())) {
-							return new KeyStoreUrlConnection(url,
-									SslStoreProviderUrlStreamHandlerFactory.this.sslStoreProvider
-											.getTrustStore());
+							KeyStore trustStore = SslStoreProviderUrlStreamHandlerFactory.this.sslStoreProvider
+											.getTrustStore();
+							if(trustStore == null) {
+								return null
```

Comment:
I don't think `null` can be returned here. The javadoc certainly doesn't say that a `null` return is permitted.